src/west/commands.py

```diff
@@ -239,25 +239,79 @@ def die_if_no_git(self):
 
     @property
     def git_version_info(self):
-        '''Returns git version info as a tuple of ints in (major,
-        minor, patch) format, like (2, 31, 1) for git version 2.31.1.
+        '''Returns git version info as a tuple of ints, usually in
+        (major, minor, patch) format, like (2, 29, 1) for git version
+        2.29.1.
 
         Aborts the program if there is no git installed.
+
+        In rare circumstances, you may get a (major, minor) tuple,
+        like (2, 29).
         '''
+        # It's perfectly safe to compare 2-tuples against 3-tuples.
+        # For example, '(2, 29) > (2, 28, 0)' is True.
+        #
+        # Language reference on comparisons:
+        #
+        #    Collections that support order comparison are ordered
+        #    the same as their first unequal elements (for example,
+        #    [1,2,x] <= [1,2,y] has the same value as x <= y). If a
+        #    corresponding element does not exist, the shorter
+        #    collection is ordered first (for example, [1,2] < [1,2,3]
+        #    is true).
+        #
+        #    https://docs.python.org/3/reference/expressions.html#comparisons
+
         if not hasattr(self, '_git_ver'):
             self.die_if_no_git()
-            # raw_ver usually looks like '2.31.1'
-            raw_ver = self.check_output(
-                [self._git, '--version']).decode().strip().split()[-1]
-            # match just the numeric prefix, in case we get something like
-            # '2.31.1-EXTRAVERSION' in some situations. The major/minor/patch
-            # group names are purely for clarity.
-            match = re.match(
-                r'^(?P<major>\d+)\.(?P<minor>\d+)\.(?P<patch>\d+)', raw_ver)
-            self._git_ver = tuple(int(group) for group in match.groups())
+            self._git_ver = self._parse_git_version(
+                self.check_output([self._git, '--version']))
             log.dbg(f'git version: {self._git_ver}', level=log.VERBOSE_VERY)
         return self._git_ver
 
+    @staticmethod
+    def _parse_git_version(raw_version):
```

Comment:
Just curious about the extra method. It's not that much code, even less without the comments.

Reply:
> Is the extra method to ease testing? Just curious.

Yes, exactly